### PR TITLE
Fix ASIHPMixedTank crash, sizing problems, and output variable names

### PR DESCRIPTION
**Pull request by shenb**
Addresses #5845 regarding the crash with running ASIHPMixedTank.idf test file on Linux.  After the fix, diffs in running ASIHPMixedTank.idf are expected. Also fix size inconsistencies in the test file, and update the output variable names.

### CODE REVIEW

**Comment by mjwitte** [comment]
@shenb We still have this warning:

```
   ************* SizeFan: : Potential issue with equipment sizing for Fan:OnOff HPWHOUTDOORFAN
   **   ~~~   ** User-Specified Maximum Flow Rate [m3/s] = 2.26850
   **   ~~~   ** differs from Design Size Maximum Flow Rate [m3/s] = 0.26850
   **   ~~~   ** This may, or may not, indicate mismatched component sizes.
   **   ~~~   ** Verify that the value entered is intended and is consistent with other components.
```

And when this is run as an annual simulation, these warnings appear (before and after this fix):

```
   ** Warning ** Temperature out of range [-100. to 200.] (PsyPsatFnTemp)
   **   ~~~   **  Routine=CalcVarSpeedHPWH, Environment=Chicago Ohare Intl Ap IL USA TMY3 WMO#=725300, at Simulation time=01/04 14:40 - 14:45
   **   ~~~   **  Input Temperature=-107.10
   ************* Simulation Error Summary *************
   *************
   ************* ===== Recurring Error Summary =====
   ************* The following recurring error messages occurred.
   *************
   *************  ** Warning ** Temperature out of range [-100. to 200.] (PsyPsatFnTemp)
   *************  **   ~~~   **   This error occurred 28 total times;
   *************  **   ~~~   **   during Warmup 0 times;
   *************  **   ~~~   **   during Sizing 0 times.
   *************  **   ~~~   **   Max=-100.38512 C  Min=-334.376389 C
```

**Comment by mjwitte** [comment]
@shenb Also, when run as an annual simulation, the hours not met for occupied cooling are now 1292 vs 127 before this fix.  Appears that something is still wrong with the logic.

**Comment by shenb** [comment]
Mike,

Thanks for identifying this. These are two sizing problems. Attached please see an update of the example file. And I also checked it in to ASIHPvalidation, which is waiting for the pull request.

Thanks,

Bo

**Comment by Myoldmopar** [comment]
@mjwitte This is now behind develop.  Are these changes acceptable in their current form, even showing the diffs on CI, or shall I merge develop in once and let CI do it again.

Actually, since it is the evening, I'll just do the merge and push and let it run overnight.

**Comment by mjwitte** [comment]
@Myoldmopar No, this is not ready - see above comments.  @shenb Attachents to github replies do not come through, and I do not see any idf file changes in this branch.  Perhaps the chance didn't get pushed up to github?

**Comment by shenb** [comment]
Mike,

Thanks for help checking. There appear deeper logic mismatching hidden in the annual simulation. I will work on this full time next week to see if we can fix it.

Bo

**Review by mjwitte — CHANGES_REQUESTED** [review]
These results are much better now.  One more thing that I noticed from the new output variables added to the test file.  The names are too generic.

```
HVAC,Average,Air Loop Flow Rate [kg/s]
HVAC,Average,Condenser Water Flow Rate [kg/s]
HVAC,Average,Heating Coil Total Air Heating Rate [W]
HVAC,Average,Total Water Heating Rate [W]
HVAC,Average,Total Electric Power [W]
HVAC,Average,Total Latent Cooling Rate [W]
HVAC,Average,Total Source Energy Rate [W]
HVAC,Average,Total COP []
HVAC,Sum,Total Electric Energy [J]
HVAC,Sum,Total Cooling Energy [J]
HVAC,Sum,Total Air Heating Energy [J]
HVAC,Sum,Total Water Heating Energy [J]
HVAC,Sum,Total Latent Cooling Energy [J]
HVAC,Sum,Total Source Energy [J]
```

All of these need common prefix to show that they come from the integrated heat pump.  I would suggest "Integrated Heat Pump . . . " this impacts source and docs.  Also, remove total, and a few other changes to be consistent with other output variables.  I should have caught this back in the initial pull request.  Suggested names:

```
HVAC,Average,Integrated Heat Pump Air Loop Mass Flow Rate [kg/s]
HVAC,Average,Integrated Heat Pump Condenser Water Mass Flow Rate [kg/s]
HVAC,Average,Integrated Heat Pump Heating Coil Air Heating Rate [W]
HVAC,Average,Integrated Heat Pump Water Heating Rate [W]
HVAC,Average,Integrated Heat Pump Electric Power [W]
HVAC,Average,Integrated Heat Pump Latent Cooling Rate [W]
HVAC,Average,Integrated Heat Pump Source Heat Transfer Rate [W]
HVAC,Average,Integrated Heat Pump COP []
HVAC,Sum,Integrated Heat Pump Electric Energy [J]
HVAC,Sum,Integrated Heat Pump Cooling Energy [J]
HVAC,Sum,Integrated Heat Pump Air Heating Energy [J]
HVAC,Sum,Integrated Heat Pump Water Heating Energy [J]
HVAC,Sum,Integrated Heat Pump Latent Cooling Energy [J]
HVAC,Sum,Integrated Heat Pump Source Heat Transfer Energy [J]
```

**Review by mjwitte — CHANGES_REQUESTED** [review]
And one more that overlapped with the one from the cooling coil, should keep the "Total" if it is truly total=sensible+latent
`Integrated Heat Pump Cooling Coil Total Cooling Rate`

Looking more at the "Rate" variables vs "Energy" variables.  If they represent the same quantity just one as W and the other as J, then the names should be the same.  Right now there is

```
Integrated Heat Pump Heating Coil Air Heating Rate
Integrated Heat Pump Air Heating Energy
```

Unless they represent a different heat flow, then they should be identical except for the last work "Rate" and "Energy".  Some others have the same issue.

File: src/EnergyPlus/IntegratedHeatPump.cc [hunk 1]
```diff
@@ -198,18 +198,19 @@ namespace EnergyPlus {
...
 						                       CompOp, 0.0, 1, 0.0, 0.0, 0.0, OnOffAirFlowRat );
+						SimVariableSpeedCoils( BlankString, IntegratedHeatPumps( DXCoilNum ).DWHCoilIndex,
+							CyclingScheme, MaxONOFFCyclesperHour, HPTimeConstant, FanDelayTime,
+							CompOp, 0.0, 1, 0.0, 0.0, 0.0, OnOffAirFlowRat);
 
 						SimVariableSpeedCoils( BlankString, IntegratedHeatPumps( DXCoilNum ).SCCoilIndex,
-						                       CyclingScheme, MaxONOFFCyclesperHour, HPTimeConstant, FanDelayTime,
-						                       CompOp, PartLoadFrac,
-						                       SpeedNum, SpeedRatio, SensLoad, LatentLoad, OnOffAirFlowRat );
+							CyclingScheme, MaxONOFFCyclesperHour, HPTimeConstant, FanDelayTime,
```
Comment (mjwitte) [inline src/EnergyPlus/IntegratedHeatPump.cc]: @shenb It appears that these changes are just indentation?  Several others like that below.

**Comment by shenb** [comment]
Thanks Mike.

I will have these changed.

**Comment by shenb** [comment]
And one more that overlapped with the one from the cooling coil, should keep the "Total" if it is truly total=sensible+latent
Integrated Heat Pump Cooling Coil Total Cooling Rate

This is true.

**Review by mjwitte — APPROVED** [review]

**Pull request merged**